### PR TITLE
Additional Network Interface Support

### PR DESCRIPTION
**Pull request by stenalpjolly**

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
modules/umig/variables.tf
--- old/modules/umig/variables.tf
+++ new/modules/umig/variables.tf
@@ -42,13 +42,13 @@
 
 variable "additional_networks" {
   description = "Additional network interface details for GCE, if any."
-  default = []
-  type        = list(object({
+  default     = []
+  type = list(object({
     network            = string
     subnetwork         = string
     subnetwork_project = string
     network_ip         = string
-    access_config      = list(object({
+    access_config = list(object({
       nat_ip       = string
       network_tier = string
     }))
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
modules/instance_template/variables.tf
--- old/modules/instance_template/variables.tf
+++ new/modules/instance_template/variables.tf
@@ -155,13 +155,13 @@
 
 variable "additional_networks" {
   description = "Additional network interface details for GCE, if any."
-  default = []
-  type        = list(object({
+  default     = []
+  type = list(object({
     network            = string
     subnetwork         = string
     subnetwork_project = string
     network_ip         = string
-    access_config      = list(object({
+    access_config = list(object({
       nat_ip       = string
       network_tier = string
     }))
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
 ```

File: modules/instance_template/variables.tf [hunk 1]
```diff
@@ -153,6 +153,21 @@ variable "network_ip" {
...
 
+variable "additional_networks" {
+  description = "Additional network interface details for GCE, if any."
+  default     = []
+  type = list(object({
+    network            = string
+    subnetwork         = string
+    subnetwork_project = string
+    network_ip         = string
+    access_config = list(object({
```
Comment (morgante) [inline modules/instance_template/variables.tf:164]: I'm not sure we should make this a list of objects—in practice, only one `access_config` is used. Can we uplevel `nat_ip` and `network_tier`?

**Review by morgante — CHANGES_REQUESTED** [review]
Thanks for working on this.

File: modules/instance_template/variables.tf (hunk 1 above)
Comment (stenalpjolly) [reply modules/instance_template/variables.tf:164]: @morgante I understand your point and what saying is right, but I want to keep the access_config as a list for the following reasons

1. According to the [Compute Insert Spec](https://cloud.google.com/compute/docs/reference/rest/v1/instances/insert#:~:text=string%2C%0A%20%20%20%20%20%20%22name%22%3A%20string%2C-,%22accessConfigs%22,-%3A%20%5B%0A%20%20%20%20%20%20%20%20%7B%0A%20%20%20%20%20%20%20%20%20%20%22type%22%3A%20enum%2C%0A%20%20%20%20%20%20%20%20%20%20%22name), the access config is expected as a list
2. [Existing Access Config](https://github.com/terraform-google-modules/terraform-google-vm/blob/master/modules/instance_template/variables.tf#L217) data type is a list. I think there should be a consistency in the data structure in all places. Users should not be confused with a list data structure in one place and object data structure in another place

**Comment by kevin-liangit** [comment]
Hey,stenalpjolly

it seems that access_config, under additional networks is mandetory. Can we make this optional?

**Pull request merged**